### DISCUSSION TITLE
`vercel dev` picks up `rewrites` even though it's not serving the serverless function

### DISCUSSION THREAD

**Discussion by jensmeindertsma** [vercel/vercel#7100] [Help] [answered]
I'm trying to deploy my Remix project to Vercel. As a part of that, I've created a serverless function at `./api/index.ts` that serves the Remix assets. When I run `vercel dev`, this correctly serves the serverless function for all requests. It also works this way in production. 

Then I try to move the serverless function to `./api/src/index.ts`. This just works in `vercel dev` because `vercel dev` invokes `remix dev` under the hood and thus doesn't have anything to do with the serverless function. When I deploy however, the serverless function is used and the function is only served at `my-domain.vercel.app/api/src/index`.

I then tried to fix this by adding a `rewrites` entry to the `vercel.json` file:
```json
{
  "rewrites": [
    {
      "source": "/(.*)",
      "destination": "/api/src/index"
    }
  ]
}
```

This fixes the problem in production. But now `vercel dev` is picking up these rewrites, even though it isn't using the serverless function at all. It thus tries to rewrite the traffic to a non-existant route and things fail.

-> How do I make `vercel dev` ignore the `vercel.json` rewrites? Or how do I make the `vercel` production environment apply a rewrite only in that environment?

**Comment by leo** [comment, marked as answer]
Thank you for reaching out!

I shared a more detailed response on the [Remix Discord](https://discord.com/channels/770287896669978684/778004202282287104/915911689826877500).

**Reply by keen99** [reply]
too bad your answer was lost to the community at large. :(

**Comment by astanciu** [comment]
Can we share the answer here instead of deep linking into discord?
